onnx/defs/tensor/defs.cc

```diff
@@ -103,6 +104,101 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
+static const char* CastLike_ver15_doc = R"DOC(
+The operator casts the elements of a given input tensor (the first input) to
+the same data type as the elements of the second input tensor.
+See documentation of the Cast operator for further details.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+	CastLike,
+	15,
+	OpSchema()
+	.SetDoc(CastLike_ver15_doc)
+	.Input(
+		0,
+		"input",
+		"Input tensor to be cast.",
+		"T1",
+		OpSchema::Single,
+		true,
+		1,
+		OpSchema::Differentiable)
+	.Input(
+		1,
+		"target_type",
+		"The (first) input tensor will be cast to produce a tensor of the same type as this (second input) tensor.",
+		"T2",
+		OpSchema::Single,
+		true,
+		1,
+		OpSchema::NonDifferentiable)
+	.Output(
+		0,
+		"output",
+		"Output tensor produced by casting the first input tensor to have the same type as the second input tensor.",
+		"T2",
+		OpSchema::Single,
+		true,
+		1,
+		OpSchema::Differentiable)
+	.TypeConstraint(
+		"T1",
+		{ "tensor(float16)",
```

Comment:
use OpSchema::all_tensor_types_with_bfloat() instead? for better readability

Reply:
That includes the complex types, but cast doesn't support complex. I have copied the same types as in cast.